lab-saul/lib/server.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+// node modules (none needed in this app)
+// npm modules
+const express = require('express');
+const mongoose = require('mongoose');
+// app modules
+// module logic
+// configure mongoose
+mongoose.Promise = Promise;
+mongoose.connect(process.env.MONGODB_URI);
+
+let server;
+const app = express();
+// load middleware
+// load routes
+app.get('/api/hello', (req, res, next) => {
+  res.send('hello world');
+});
+
+app.use(require('../route/car-store-router.js'));
+
+// loader err handler
+app.use((err, req, res, next) => {
+  console.log('err', err);
+  if(!err){
```

Comment:
Mongo is sending back an error when you try to find a document that doesn't exist. Doesn't look like you're setting up a way to convert that into a 404 'not found' error anywhere. Find me tomorrow if you have any questions about setting that up.